icu4j/main/classes/core/src/com/ibm/icu/impl/OlsonTimeZone.java

```diff
@@ -1289,8 +1283,6 @@ public TimeZone freeze() {
     public TimeZone cloneAsThawed() {
         OlsonTimeZone tz = (OlsonTimeZone)super.cloneAsThawed();
         if (finalZone != null) {
-            // TODO Do we really need this?
-            finalZone.setID(getID());
```

Comment:
This line was the source of the thread sanitizer failures. It's modifying the source of the clone, which shouldn't be done. Concurrent clones caused concurrent setID()s, which were flagged as a problem.

I will speculate that the finalZone.setID() may have been added here to work around it not having been set when the OlsonTimezone was originally constructed, line 484. Every other path seems to keep the finalZone ID in sync with the main ID.